src/shared/src/components/molecules/AltinnModal.tsx

```diff
@@ -98,66 +92,53 @@ const styles = createStyles({
-                    classes.iconBtn,
-                    { [classes.closeButtonOutsideModal]: this.props.closeButtonOutsideModal === true},
-                  )}
-                  onClick={this.props.onClose}
-                >
-                  <i tabIndex={0} className={classNames('ai ai-exit-test', classes.iconStyling)} />
-                </IconButton>
-              }
-              <Typography className={classes.headerText}>
-                {this.props.headerText}
-              </Typography>
-            </div>
-            <div className={classes.body}>
-              {this.props.children}
-            </div>
-          </div>
-        </Modal>
-      );
-    } else {
-      return (
-        <div className={classes.modal}>
-          <div className={classes.header}>
-            {this.props.hideCloseIcon && this.props.hideCloseIcon === true ? null :
-              <IconButton
-                className={classNames(
-                  classes.iconBtn,
-                  { [classes.closeButtonOutsideModal]: this.props.closeButtonOutsideModal === true},
-                )}
-                onClick={this.props.onClose}
-              >
-                <i tabIndex={0} className={classNames('ai ai-exit-test', classes.iconStyling)} />
-              </IconButton>
-            }
-            <Typography className={classes.headerText}>
-              {this.props.headerText}
-            </Typography>
-          </div>
-          <div className={classes.body}>
-            {this.props.children}
-          </div>
-        </div>
-      );
-    }
+const useStyles = makeStyles(styles);
+
+const StyledModal = (props: React.PropsWithChildren<IAltinnModalComponentProvidedProps>) => {
+  const classes = useStyles();
+  return (
+    <div className={classes.modal}>
+      <div className={classes.header}>
+        {props.hideCloseIcon && props.hideCloseIcon === true ? null :
+          <IconButton
+            className={classNames(
+              classes.iconBtn,
+              {[classes.closeButtonOutsideModal]: props.closeButtonOutsideModal === true},
+            )}
+            onClick={props.onClose}
+          >
+            <i className={classNames('ai ai-exit-test', classes.iconStyling)} tabIndex={-1}/>
+          </IconButton>
+        }
+        <Typography className={classes.headerText}>
+          {props.headerText}
+        </Typography>
+      </div>
+      <div className={classes.body}>
+        {props.children}
+      </div>
+    </div>
+  );
+}
+
+const AltinnModal = (props: React.PropsWithChildren<IAltinnModalComponentProvidedProps>) => {
+  const classes = useStyles();
+  if (!props.printView) {
+    const {isOpen, hideBackdrop, allowCloseOnBackdropClick, onClose} = props;
+    return (
+      <Modal
+        open={isOpen}
+        className={classes.scroll}
+        hideBackdrop={hideBackdrop}
+        onBackdropClick={allowCloseOnBackdropClick === false ? null : onClose}
+      >
+        <StyledModal {...props}/>
```

Comment:
I would suggest here as well to destruct immediately:

```suggestion
const AltinnModal = ({isOpen, hideBackdrop, allowCloseOnBackdropClick, onClose, printView, ...rest}: React.PropsWithChildren<IAltinnModalComponentProvidedProps>) => {
  const classes = useStyles();
  if (!printView) {
    return (
      <Modal
        open={isOpen}
        className={classes.scroll}
        hideBackdrop={hideBackdrop}
        onBackdropClick={allowCloseOnBackdropClick === false ? null : onClose}
      >
        <StyledModal onClose={onClose} {...rest}/>
```

Reply:
In this case I'm not sure if I want to move so much at this point.
I will then have write the unit-tests first, so I know that all args are properly included in both branches.

Reply:
We also know what props `StyledModal` needs. But I agree its not really clear - this is one of the downsides of just spreading props like this :) 

Adding more tests to this is a good idea though, and that would make refactoring safer.